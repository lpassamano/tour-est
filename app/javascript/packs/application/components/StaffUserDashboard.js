import React, { Component } from "react";
import CreateTourForm from "./CreateTourForm";

class StaffUserDashboard extends Component {
  async componentDidMount() {
    const { ok, data } = await this.props.onAuthenticate();
    if (!ok) {
      throw new Error("Not Authenticated!");
<<<<<<< HEAD
=======
    } else {
      this.setState({
        username: this.props.currentStaffUser.username,
        id: this.props.currentStaffUser.id,
        cultural_center: {
          name: this.props.currentStaffUser.cultural_center.name,
          // not recieving cc id in the props!!
          id: this.props.currentStaffUser.cultural_center.id
        }
      });
>>>>>>> Add snapshot test
    }
  }

  render() {
    const { username, cultural_center, id } = this.props.currentStaffUser;
    return (
      <div>
        <h1>Staff Dashboard</h1>
        <h3>Current user: {username}</h3>
        <h3>{cultural_center.name}</h3>
        <CreateTourForm
          onCreateTour={this.props.onCreateTour}
          staffUserId={id}
          culturalCenterId={cultural_center.id}
        />
      </div>
    );
  }
}

export default StaffUserDashboard;
