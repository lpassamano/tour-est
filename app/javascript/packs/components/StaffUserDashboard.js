import React, { Component } from 'react'
import { create } from 'apisauce'

const api = create({
  baseURL: '/'
})

class StaffUserDashboard extends Component {
  state = {
    username: "",
    id: "",
    cultural_center: {
      name: "",
      id: ""
  }}

  async componentDidMount() {
    const token = window.localStorage.getItem('token');
    api.setHeader('Authorization', `Token token="${token}"`);
    const { ok, data } = await api.get('/staff_user');

    if (!ok) {
      throw new Error('Not Authenticated!');
    }

    console.log('user is', data);
  }

  render() {
    return (
      <div>
        Staff Dashboard
      </div>
    )
  }
}



export default StaffUserDashboard;
