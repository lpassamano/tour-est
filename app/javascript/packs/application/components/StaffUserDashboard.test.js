import StaffUserDashboard from "./StaffUserDashboard";
import React from "react";
import { mount } from "enzyme";

describe("<StaffUserDashboard />", () => {
  test("render", () => {
    const onCreateTour = jest.fn();
    const currentStaffUser = {
      username: "username123",
      id: 1,
      cultural_center: { name: "Penn Museum", id: 1 }
    };
    const tours = {
      isFetching: false,
      data: [
        { id: 1, title: "Tour 1" },
        { id: 2, title: "Tour 2" },
        { id: 3, title: "Tour 3" }
      ]
    };
    const component = mount(
      <StaffUserDashboard
        onCreateTour={onCreateTour}
        currentStaffUser={currentStaffUser}
        tours={tours}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
