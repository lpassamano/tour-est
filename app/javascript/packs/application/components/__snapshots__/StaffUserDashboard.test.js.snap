// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffUserDashboard /> render 1`] = `
<StaffUserDashboard
  currentStaffUser={
    Object {
      "cultural_center": Object {
        "id": 1,
        "name": "Penn Museum",
      },
      "id": 1,
      "tours": Array [
        Object {
          "id": 1,
          "title": "Tour 1",
        },
        Object {
          "id": 2,
          "title": "Tour 2",
        },
        Object {
          "id": 3,
          "title": "Tour 3",
        },
      ],
      "username": "username123",
    }
  }
  onCreateTour={[MockFunction]}
>
  <div>
    <h1>
      Staff Dashboard
    </h1>
    <h3>
      Current user: 
      username123
    </h3>
    <h3>
      Penn Museum
    </h3>
    <ToursList
      tours={
        Array [
          Object {
            "id": 1,
            "title": "Tour 1",
          },
          Object {
            "id": 2,
            "title": "Tour 2",
          },
          Object {
            "id": 3,
            "title": "Tour 3",
          },
        ]
      }
    >
      <div>
        <TourInfo
          key="1"
          title="Tour 1"
        >
          <div>
            <h4>
              Tour 1
            </h4>
          </div>
        </TourInfo>
        <TourInfo
          key="2"
          title="Tour 2"
        >
          <div>
            <h4>
              Tour 2
            </h4>
          </div>
        </TourInfo>
        <TourInfo
          key="3"
          title="Tour 3"
        >
          <div>
            <h4>
              Tour 3
            </h4>
          </div>
        </TourInfo>
      </div>
    </ToursList>
    <CreateTourForm
      culturalCenter={1}
      navigate={[Function]}
      onCreateTour={[MockFunction]}
      staffUser={1}
    >
      <form
        onSubmit={[Function]}
      >
        <input
          id="title"
          name="title"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Create Tour
        </button>
      </form>
    </CreateTourForm>
  </div>
</StaffUserDashboard>
`;
